src/tasks/FileUploadUtil/Interfaces/IProgress.ts

```diff
@@ -0,0 +1,26 @@
+/* eslint-disable @typescript-eslint/type-annotation-spacing */
+import { GraphClientError } from "../../../GraphClientError";
+import { Range } from "../../../Range";
+import { UploadResult } from "../UploadResult";
+
+/**
+ * Interface enabling progress handling with callbacks.
+ */
+export interface Progress {
```

Comment:
@zengin  How about `UploadProgressHandler`? 

`IProgressCallback`   and `IProgress` is use in Java and C#. So I thought of aligning close to that.
TypeScript does not use `I-` interface naming.

Reply:
Having `Progress` in the name is misleading IMO because it has an event handler defined inside this object for `Progress`. How about `UploadEventHandlers`?

`Progress` can also be used to represent the "full progress" as intended in the current state of the naming (e.g. `failure`, `uploading`, `uploaded` states in the progression of upload), but then we need to find a different name for that `uploading` state other than `progress`. Because it causes us to have calls like: `progressCallBack.progress`, similar to `error.error` we have discussed in other PRs.

Reply:
Done